src/legacy/state/orders/updaters/GpOrdersUpdater.ts

```diff
@@ -41,7 +42,7 @@ const statusMapping: Record<OrderTransitionStatus, OrderStatus | undefined> = {
 }
 
 function _transformGpOrderToStoreOrder(
-  order: EnrichedOrder,
+  { order, composableCowInfo }: OrderWithComposableCowInfo,
```

Comment:
Based on the suggestion in the type definition, you won't need to de-scrutcure this.
Just change the type should be enough, as the original order is spread onto the `storeOrder` obj

Reply:
I don't think we should mix interfaces.
`EnrichedOrder` represents an order model from SDK, if we mix something in this interface it will be pretty difficult to distinguish which property is from SDK and which is local.
We already have some troubles because of violation of Interface segregation principle:
https://github.com/cowprotocol/cowswap/pull/2596

Reply:
Ok, that's fair.

Then I think the name `order` for this structure is not accurate, because it has the order as property, but it's not an order object.
For example, when working with instances of this type, we'll have a `order` where the actual order is accessed via `order.order`.

What about naming it something like: `ComposableCowOrder` - although that is also named `Order` which goes against my point...
What about `ComposableCowObj`?
Still not happy with those names, please let me know if you have any better idea

Reply:
So, do you suggest renaming of `OrderWithComposableCowInfo` to `ComposableCowOrder` or `ComposableCowObj `? If yes, I'm ok with it

Reply:
Yes, that's right.
I'm just not convinced what to name this structure

Reply:
Technically, the main entity in this structure is an order (`EnrichedOrder`) from SDK. `ComposableCowInfo` it's optional and additional data.

Now there are three layers of orders:
1. [`EnrichedOrder`](https://github.com/cowprotocol/cow-sdk/blob/main/src/order-book/types.ts#L3) - DTO from SDK
2. `OrderWithComposableCowInfo` - `EnrichedOrder` + `ComposableCowInfo`
3.  [`Order`](https://github.com/cowprotocol/cowswap/blob/15664530daca007c9b67e89e32d979ad06591550/src/legacy/state/orders/actions.ts#L111) - it's also `EnrichedOrder` but legacy one + some app-level data (eth-flow, tokens, etc)

I think the last two layers must be merged into something like `CowSwapOrder` or `AppOrder`.
The idea is:
 - a collection of "raw"/DTO orders from SDK: `EnrichedOrder[]`
 - a collection of app-level orders: `CowSwapOrder[]`
 These two collections should be linked by id one-to-one and don't duplicate data together.
 
 @alfetopito Is it ok if I create an issue to refactor it in the future?
 /cc @anxolin

Reply:
I'm ok with that

Reply:
Issue: https://github.com/cowprotocol/cowswap/issues/2648